### PR TITLE
[Dy2Stat]Fix losting pre/post hook from outermost layer while jit.save

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

Bug fixes
### PR changes

Others
### Describe

[Dy2Stat]Fix losting pre/post hook from outermost layer while jit.save

### CODE REVIEW

**Comment by rainyfly** [comment]
当放入input_spec参数时，程序会报错

**Comment by rainyfly** [comment]
Works, LGTM

File: python/paddle/fluid/dygraph/jit.py [hunk 1]
```diff
@@ -830,10 +836,16 @@ def fun(inputs):
...
+    # whether outermost layer has pre/post hook, if does, we need also save
+    # these operators in program. 
+    with_hook = configs.with_hook
+
     scope = core.Scope()
     extra_var_info = dict()
     if isinstance(layer, Layer):
         functions = dir(inner_layer)
+        if inner_layer._forward_pre_hooks or inner_layer._forward_post_hooks:
+            with_hook = True
```
Comment (0x45f) [inline python/paddle/fluid/dygraph/jit.py:848]: 后面可以在讨论下with_hook参数的是否必要

**Review by 0x45f — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/jit.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/jit.py:848]: 好的

**Pull request merged**